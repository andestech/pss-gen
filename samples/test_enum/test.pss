component pss_top {
	enum count_e {
                COUNT_1 = 1,
		COUNT_2, COUNT_3, COUNT_4,
		COUNT_10 = 10,
		COUNT_NEG = -1
        };

	action printf {
		rand count_e val = 0;
		exec body ASM =
		"""count_e = {{val}}""";
	}
	action root_a {
		activity {
			do printf with {val == COUNT_1;};
			do printf with {val == COUNT_2;};
			do printf with {val == COUNT_3;};
			do printf with {val == COUNT_4;};
			do printf with {val == COUNT_10;};
			do printf with {val == COUNT_NEG;};
		}
	}

}

