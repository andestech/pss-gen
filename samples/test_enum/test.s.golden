
inVal_0 = 1
inVal_1 = -1
inVal_2 = 20
outVal_0 = 4
outVal_1 = 50
outVal_2 = -1
IN_2 = 2
OUT_0 = 0
OUT_10 = 10
SAME_NAME = 20
out_e::SAME_NAME = 50
in_e::SAME_NAME = 20


outVal = 3
inVal = 4
outVal = 4
inVal = 2
